ballerina/engine.bal

```diff
@@ -62,16 +62,26 @@ isolated class Engine {
         return getOutputObjectFromErrorDetail(errorDetail);
     }
 
-    isolated function validateDocument(parser:DocumentNode document) returns OutputObject? {
+    isolated function validateDocument(parser:DocumentNode document, map<json>? variables) returns OutputObject? {
+        FragmentVisitor fragmentVisitor = new(document);
```

Comment:
Shall we move the `operationNode.getErrors()` check from `ValidatorVisitor` to here?

Reply:
We have to come up with a way to handle the errors in the operation nodes as well.